components/assignmentList.js

```diff
@@ -0,0 +1,200 @@
+import { router } from "expo-router";
+import { useMemo } from "react";
+import { SectionList, TouchableOpacity, View } from "react-native";
+import { Image } from "react-native-expo-image-cache";
+import { Text } from "react-native-paper";
+import { themeColors } from "~/Constants";
+import { formatDate } from "~/Utility";
+import AssignmentCard from "~/components/assignmentCard";
+import EmptyScreen from "~/components/emptyScreen";
+import RefreshInvalidate from "~/components/refreshInvalidate";
+
+const AssignmentsList = ({
+  role,
+  playerInfo,
+  userInfo,
+  invalidateKeys,
+  drillInfo,
+  children,
+}) => {
+  const today = formatDate(Date.now());
+
+  let assigned_data = userInfo ? userInfo.assigned_data : [];
+
+  if (!userInfo) {
+    const alreadyAddedData = {};
+
+    Object.values(playerInfo).forEach((player) => {
+      player["assigned_data"].forEach((assignment) => {
+        const { assignedTime, drillId, completed, attemptId } = assignment;
+        const { uid, pfp, name: userName } = player;
+
+        if (!alreadyAddedData[assignedTime]) {
+          alreadyAddedData[assignedTime] = {};
+        }
+
+        if (!alreadyAddedData[assignedTime][drillId]) {
+          alreadyAddedData[assignedTime][drillId] = {
+            assignedTime,
+            drillId,
+            players: [],
+          };
+        }
+
+        alreadyAddedData[assignedTime][drillId].players.push({
+          pfp,
+          userName,
+          uid,
+          completed,
+          attemptId,
+        });
+      });
+    });
+
+    Object.values(alreadyAddedData).forEach((assignedSortedByTime) => {
+      Object.values(assignedSortedByTime).forEach((assignedSortedByDrillId) => {
+        assigned_data.push(assignedSortedByDrillId);
+      });
+    });
+  }
+
+  // Group the assigned drills by date
+  const groupedData = useMemo(() => {
+    return assigned_data.reduce((acc, curr) => {
+      const date = formatDate(curr.assignedTime);
+      const dateKey = date === today ? "Today" : date;
+
+      if (!acc[dateKey]) {
+        acc[dateKey] = [];
+      }
+      if (curr.completed) {
+        acc[dateKey].push(curr);
+      } else {
+        acc[dateKey].unshift(curr);
+      }
+
+      return acc;
+    }, {});
+  }, [assigned_data, today]);
+
+  // Sort the dates in descending order
+  const sortedDates = useMemo(() => {
+    return Object.keys(groupedData).sort((a, b) => new Date(b) - new Date(a));
+  }, [groupedData]);
+
+  if (sortedDates.length === 0) {
+    return (
+      <EmptyScreen
+        invalidateKeys={invalidateKeys}
+        text={"No drills assigned"}
+      />
+    );
+  }
+
+  const stackedPfp = (playerList) => {
+    const numPfp = Math.min(3, playerList.length);
+
+    const pfpArr = playerList.slice(0, numPfp).map((player, index) => {
```

Comment:
Are we able to make it display 3 image cards and then the last card displays "+{numAssignedPlayers - 3}" if there are more than 3 players assigned?